Tween/DelegateAnimation.cs

```diff
@@ -103,5 +143,68 @@ public override void Resume()
 				Debug.LogError("DelegateAnimation: You cannot resume this animation because you haven't started your delegate animation yet!");
 			}
 		}
+
+		public void ScaleTime(float duration)
+		{
+			if (Animation == null)
+			{
+				speedChangeRequest = () => ScaleInnerAnimationTime(duration);
+			}
+			else
+			{
+				ScaleInnerAnimationTime(duration);
+			}
+		}
+
+		public void ChangeSpeed(float multiplier)
+		{
+			if (Animation == null)
+			{
+				speedChangeRequest = () => ChangeInnerAnimationSpeed(multiplier);
+			}
+			else
+			{
+				ChangeInnerAnimationSpeed(multiplier);
+			}
+		}
+
+		public void Reverse()
+		{
+			if (Animation == null)
+			{
+				reverseChangeRequest = ReverseInnerAnimation;
+			}
+			else
+			{
+				ReverseInnerAnimation();
+			}
+		}
+
+		private void ScaleInnerAnimationTime(float duration)
+		{
+			var playbackControl = Animation as IAnimationPlaybackControl;
+			if (playbackControl != null)
+			{
+				playbackControl.ScaleTime(duration);
+			}
+		}
+
+		private void ChangeInnerAnimationSpeed(float multiplier)
```

Comment:
You might find out a bit DRY for these functions but it has better performance than using Reflections, callbacks, etc.

Let me know if you can reduce the lines by not using those tricks.